src/datadog/tracer.cpp

```diff
@@ -36,6 +36,7 @@ Tracer::Tracer(const FinalizedTracerConfig& config)
 Tracer::Tracer(const FinalizedTracerConfig& config,
                const std::shared_ptr<const IDGenerator>& generator)
     : logger_(config.logger),
+      config_manager_(std::make_shared<ConfigManager>(config)),
```

Comment:
I moved this up so that it occurs before `collector_`. This is not at all necessary, since we're talking about shared pointers, but it mirrors what we would do if the collector and the config manager were both held by value.